src/Umbraco.Web.Website/ActionResults/RedirectToUmbracoPageResult.cs

```diff
@@ -122,8 +123,17 @@ public Task ExecuteResultAsync(ActionContext context)
             }
 
             HttpContext httpContext = context.HttpContext;
-            IIOHelper ioHelper = httpContext.RequestServices.GetRequiredService<IIOHelper>();
-            string destinationUrl = ioHelper.ResolveUrl(Url);
+
+            string destinationUrl;
+            if (IsUrlAbsolute(Url))
+            {
+                destinationUrl = Url;
+            }
+            else
+            {
+                IHostingEnvironment hostingEnvironment = httpContext.RequestServices.GetRequiredService<IHostingEnvironment>();
+                destinationUrl = hostingEnvironment.ToAbsolute(Url);
+            }
```

Comment:
Using the ASP.NET Core built-in [`IUrlHelper.Content()`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.iurlhelper.content?view=aspnetcore-5.0) seems more appropriate, as we only want to convert virtual paths (starting with `~`) to be resolved to the application absolute path:

```suggestion
            var urlHelper = httpContext.RequestServices.GetRequiredService<IUrlHelper>();
            string destinationUrl = urlHelper.Content(Url);
```

This also removes the dependency on `IHostingEnvironment`, which we [might want to remove in the future](https://github.com/umbraco/Umbraco-CMS/pull/11990#issuecomment-1045287496).